internal/cloud/run.go

```diff
@@ -483,33 +497,43 @@ func NewRunService(tfe *tfe.Client) RunService {
 	return &runService{tfe}
 }
 
-type CreateRunOptions struct {
-	Organization           string
-	Workspace              string
-	ConfigurationVersionID string
-	Message                string
-	PlanOnly               bool
-	RunVariables           []*tfe.RunVariable
-}
+func getDesiredRunStatus(run *tfe.Run, policyChecksEnabled bool, costEstimateEnabled bool) []tfe.RunStatus {
+	// shared desired status across all runs
+	desiredStatus := []tfe.RunStatus{
+		tfe.RunPolicySoftFailed,
+		tfe.RunPlannedAndFinished,
+		tfe.RunApplied,
+	}
 
-type ApplyRunOptions struct {
-	RunID   string
-	Comment string
-}
+	// when plan_only run
+	if run.PlanOnly {
+		// plan only runs will result in default desired slice
+		// most likely planned_and_finished or policy_soft_failed
+		return desiredStatus
+	}
 
-type GetRunOptions struct {
-	RunID string
-}
+	// when auto_apply run
+	if run.AutoApply {
+		if policyChecksEnabled {
+			// policy override requies human approval before proceeding, run has reached no-op
+			desiredStatus = append(desiredStatus, tfe.RunPolicyOverride)
+		}
+		return desiredStatus
+	}
 
-type DiscardRunOptions struct {
-	RunID   string
-	Comment string
-}
+	// when applyable/confirmable run
+	// determine which various run status it can end with
+	if !run.PlanOnly && !run.AutoApply {
+		if costEstimateEnabled && !policyChecksEnabled {
```

Comment:
This might be easier to read if we append to `desiredStatus` independently for `costEstimateEnabled` and `policyChecksEnabled`. I might be wrong but is the evaluation for both `costEstimateEnabled` and `policyChecksEnabled` inter-related. Does doing something like this make sense? 
```
if policyChecksEnabled {
			desiredStatus = append(desiredStatus, tfe.RunPolicyChecked, tfe.RunPolicyOverride)
}

if costEstimateEnabled {
    desiredStatus = append(desiredStatus, tfe.RunCostEstimated)
}
```

Reply:
Great question. The problem is one run phase executes prior to another. If we were to add `tfe.RunCostEstimated` if Sentinel policies are also enabled; then the `run create` cmd would eject too soon.